engine/src/test/scala/cromwell/engine/workflow/lifecycle/execution/job/preparation/JobPreparationActorSpec.scala

```diff
@@ -158,26 +159,31 @@ class JobPreparationActorSpec
     }
   }
 
-  it should "retry with more memory attribute when `memoryMultiplier` in BackendJobDescriptorKey is greater than 1" in {
-    val attributes = Map(
-      "memory" -> WomString("2 GB")
+  it should "lookup MemoryMultiplier key/value if available and accordingly update runtime attributes" in {
```

Comment:
This is a nice "golden path" test, but it would be nice to have a test to push this feature to its limits. Can we mimic a series of 100 failures, each of which requires a memory retry of 1.1? (don't worry, the numbers shouldn't get _too_ big...)
![image](https://user-images.githubusercontent.com/13006282/127211672-06a8980e-a912-4249-9ca9-ea664cc5d51c.png)

Reply:
Although having said that, knowing what happens when these exponents get out of hand might be a good idea. Eg maybe we do want to test what happens when the factor is 99.0 and we retry 100 times? Or maybe a sanity limit at 99.0 is a good idea after all? 🤷‍♂️

Reply:
I have added the test case to mimic retry 100 times. I am not sure if this is what you had in mind?